exp/controllers/machinepool_controller.go

```diff
@@ -127,6 +127,7 @@ func (r *MachinePoolReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, rete
 	}
 
 	// Initialize the patch helper.
+	// NOTE: This object doesn't implement observedGeneration. No need to add the WithStatusObservedGeneration option.
```

Comment:
Should we add ObservedGeneration to MachinePool? It seems like we would want it, similar to the other types that it was added to.

Reply:
observedGeneration to MachinePool will be added with a new PR